src/components/ai-input.tsx

```diff
@@ -28,12 +28,42 @@ export const AIInput = ( {
 }: AIInputProps ) => {
 	const inputRef = useRef< HTMLTextAreaElement >( null );
 
+	const [ isTyping, setIsTyping ] = useState( false );
+	const [ typingTimeout, setTypingTimeout ] = useState< NodeJS.Timeout | null >( null );
+
+	const { RiveComponent, inactiveInput, thinkingInput, typingInput, startStateMachine } =
+		useRiveIcon();
+
 	useEffect( () => {
 		if ( ! disabled && inputRef.current ) {
 			inputRef.current.focus();
 		}
 	}, [ disabled ] );
 
+	useEffect( () => {
+		startStateMachine();
+
+		if ( inactiveInput ) {
```

Comment:
We pass state directly to the Rive runtime. The Rive animation has been programmed to determine which state to display based on this data so we don't have to implement that in React.

<img width="1557" alt="Screenshot 2024-06-20 at 16 38 04" src="https://github.com/Automattic/studio/assets/417538/ea471c85-5146-4690-bfc5-12fbc7968336">